From 7453ec7c41f500169506a126fe8dd9ea23a92a03 Mon Sep 17 00:00:00 2001
From: Andrew Hopkins <andhop@amazon.com>
Date: Mon, 24 Jul 2023 13:47:55 -0700
Subject: [PATCH] Run additional HAProxy tests (#1097)

* Install socat and lua for more haproxy unit tests
* Use recomended makefile target to run tests and add todo to use simpler target
---
 .../linux-x86/ubuntu-22.04_base/Dockerfile      |  3 ++-
 tests/ci/integration/run_haproxy_integration.sh | 17 ++++++++++-------
 2 files changed, 12 insertions(+), 8 deletions(-)

diff --git a/tests/ci/docker_images/linux-x86/ubuntu-22.04_base/Dockerfile b/tests/ci/docker_images/linux-x86/ubuntu-22.04_base/Dockerfile
index 3276cc16b..e8503bb0b 100644
--- a/tests/ci/docker_images/linux-x86/ubuntu-22.04_base/Dockerfile
+++ b/tests/ci/docker_images/linux-x86/ubuntu-22.04_base/Dockerfile
@@ -44,11 +44,12 @@ RUN set -ex && \
     libpcre2-dev \
     libreadline-dev \
     libudev-dev \
+    liblua5.4-dev \
+    socat \
     zlib1g-dev \
     dpkg-dev \
     flex \
     bison \
-    curl \
     jq \
     unzip && \
     # Based on https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2-linux.html
diff --git a/tests/ci/integration/run_haproxy_integration.sh b/tests/ci/integration/run_haproxy_integration.sh
index 8b44ea3f4..e6097a2a4 100755
--- a/tests/ci/integration/run_haproxy_integration.sh
+++ b/tests/ci/integration/run_haproxy_integration.sh
@@ -25,10 +25,14 @@ export LD_LIBRARY_PATH="${AWS_LC_INSTALL_FOLDER}/lib"
 
 function build_and_test_haproxy() {
   cd ${HAPROXY_SRC}
-  make CC="${CC}" -j ${NUM_CPU_THREADS} TARGET=generic USE_OPENSSL=1 SSL_INC="${AWS_LC_INSTALL_FOLDER}/include" SSL_LIB="${AWS_LC_INSTALL_FOLDER}/lib/"
+  make CC="${CC}" -j ${NUM_CPU_THREADS} TARGET=generic USE_OPENSSL=1 SSL_INC="${AWS_LC_INSTALL_FOLDER}/include" \
+      SSL_LIB="${AWS_LC_INSTALL_FOLDER}/lib/" USE_LUA=1  LUA_LIB_NAME=lua5.4
 
-  # These tests pass when run locally but are not supported in CodeBuild, see CryptoAlg-1965
-  excluded_tests=("mcli_show_info.vtc" "mcli_start_progs.vtc" "tls_basic_sync.vtc" "tls_basic_sync_wo_stkt_backend.vtc" "acl_cli_spaces.vtc" "http_reuse_always.vtc")
+  # These tests are marked as SLOW and should be skipped.
+  # TODO: update this to: make reg-tests VTEST_PROGRAM=../vtest/vtest REGTESTS_TYPES=default,bug,devel
+  # ssl_dh.vtc expects to use libssl with a FFDH ciphersuite which is unsupported, it will be gracefully turned off in
+  # ssl_dh.vtc with the change in https://github.com/andrewhop/haproxy/pull/1
+  excluded_tests=("mcli_show_info.vtc" "mcli_start_progs.vtc" "tls_basic_sync.vtc" "tls_basic_sync_wo_stkt_backend.vtc" "acl_cli_spaces.vtc" "http_reuse_always.vtc" "ocsp_auto_update.vtc" "ssl_dh.vtc")
   test_paths=""
 
   for test in reg-tests/**/*; do
@@ -37,7 +41,7 @@ function build_and_test_haproxy() {
       fi
   done
 
-  ./scripts/run-regtests.sh "$test_paths"
+  make reg-tests VTEST_PROGRAM=../vtest/vtest REG_TEST_FILES="$test_paths"
 }
 
 # Make script execution idempotent.
@@ -49,15 +53,14 @@ mkdir -p ${AWS_LC_BUILD_FOLDER} ${AWS_LC_INSTALL_FOLDER}
 git clone --depth 1 https://github.com/haproxy/haproxy.git
 cd haproxy
 ./scripts/build-vtest.sh
-export VTEST_PROGRAM=$(realpath ../vtest/vtest)
 
 # Test with static AWS-LC libraries
-aws_lc_build ${SRC_ROOT} ${AWS_LC_BUILD_FOLDER} ${AWS_LC_INSTALL_FOLDER} -DBUILD_SHARED_LIBS=0
+aws_lc_build ${SRC_ROOT} ${AWS_LC_BUILD_FOLDER} ${AWS_LC_INSTALL_FOLDER} -DBUILD_SHARED_LIBS=0 -DBUILD_TESTING=0
 build_and_test_haproxy $HAPROXY_SRC
 
 rm -rf ${AWS_LC_INSTALL_FOLDER}/*
 rm -rf ${AWS_LC_BUILD_FOLDER}/*
 
 # Test with shared AWS-LC libraries
-aws_lc_build ${SRC_ROOT} ${AWS_LC_BUILD_FOLDER} ${AWS_LC_INSTALL_FOLDER} -DBUILD_SHARED_LIBS=1
+aws_lc_build ${SRC_ROOT} ${AWS_LC_BUILD_FOLDER} ${AWS_LC_INSTALL_FOLDER} -DBUILD_SHARED_LIBS=1 -DBUILD_TESTING=0
 build_and_test_haproxy $HAPROXY_SRC
